knimin/lib/data_access.py

```diff
@@ -2018,6 +2020,22 @@ def getProjectNames(self):
         sql = """SELECT project FROM project"""
         return [x[0] for x in self._con.execute_fetchall(sql)]
 
+    def set_deposited_ebi(self):
+        """Updates barcode deposited status by checking EBI"""
+        accession = 'ERP012803'
+        samples = fetch_url(
+            'http://www.ebi.ac.uk/ena/data/warehouse/filereport?accession='
+            '%s&result=read_run&fields=sample_alias' % accession)
+        # Clean EBI formatted sample names to just the barcodes
+        # stripped of any appended letters for barcodes run multiple times
+        barcodes = tuple(s.strip().split('.')[1][:9]
+                         for s in samples if len(s.split('.')) == 2)
```

Comment:
Wonder if a regex would be more explicit here? Example output the resource request is below. I think a sufficient regex would be:

```python
13:20:15 (dtmcdonald@barnacle):ag_qiita_seqs_15feb2015> s = 'qiita_sid_10317:10317.000001031'

13:20:25 (dtmcdonald@barnacle):ag_qiita_seqs_15feb2015> re.compile('(\d){9}$').search(s).group(0)
Out[16]: '000001031'
```

Example text from EBI:

```text
sample_alias
qiita_sid_10317:10317.000001002
qiita_sid_10317:10317.000001004
qiita_sid_10317:10317.000001008
qiita_sid_10317:10317.000001018
qiita_sid_10317:10317.000001028
qiita_sid_10317:10317.000001031
qiita_sid_10317:10317.000001032
qiita_sid_10317:10317.000001038
```